### ISSUE TITLE
OS X Installer script doesn't work with zsh

### ISSUE THREAD

**Issue by jonaf** [NixOS/nix#4990]
**Describe the bug**

The installer for Mac OS writes this line to `~/.bash_profile`:

```
if [ -e /Users/jona/.nix-profile/etc/profile.d/nix.sh ]; then . /Users/jona/.nix-profile/etc/profile.d/nix.sh; fi # added by Nix installer
```

However, Apple switched the default shell from `bash` to `zsh` a couple years back. If you're using zsh like I am, you might restart your computer and then be unable to find `nix-shell`, and the reason is because the installer wrote the above line in `/.bash_profile` instead of `~/.zshrc`

It would be great if the installer detected the shell that was in use and update that.

**Steps To Reproduce**

1. Be a `zsh` user
2. Run the installer script from https://nixos.org/guides/install-nix.html
3. Open a new terminal session
4. Notice that `nix-env`, `nix-shell`, etc are not in the `$PATH`

**Expected behavior**

It's expected that new terminal sessions have the nix commands in the `$PATH`

**Comment by jonaf** [comment]
On closer inspection, I realized that the installer _does_ add the line in question to both `~/.bash_profile` _and_ `~/.zshrc`

however, the last line of my `~/.zshrc` was a commented line at the time of the install, and it looks like I didn't have a newline at the end of that line, so the installer just appended to the last line instead of adding a new line. This effectively commented it out from the installer, so this issue is a fairly specific situation (where a person like me doesn't properly terminate a file's contents with a newline).

**Comment by domenkozar** [comment]
That's fixed in master, but needs backporting to 2.3 /cc @toonn

**Comment by toonn** [comment]
I believe this can be closed. The backport isn't necessary anymore since 2.4 has been released.
